ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java

```diff
@@ -86,12 +93,18 @@ public Configuration getConf() {
 
   public void init(AtomicBoolean stop) throws Exception {
     setPriority(MIN_PRIORITY);
-    setDaemon(true); // this means the process will exit without waiting for this thread
+    setDaemon(false);
     this.stop = stop;
     this.hostName = ServerUtils.hostname();
     this.runtimeVersion = getRuntimeVersion();
   }
 
+  protected void checkInterrupt() throws InterruptedException {
+    if (Thread.interrupted()) {
+      throw new InterruptedException(type.name() + " execution is interrupted.");
```

Comment:
Since we already have class hierarchy to define common and specific parts of the code the `CompactorThreadType` enum is not necessary. Especially if it is used only for naming subclasses.

I think a method like
```
abstract String getName(); 
``` 
and the corresponding implementations in each subclass can be used.

Or maybe something like `this.getClass().getName()` also works.